src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/Waiting/VisualStudioOperationWaitIndicator.cs

```diff
@@ -0,0 +1,122 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+
+using Microsoft.VisualStudio.ProjectSystem.Waiting;
+using Microsoft.VisualStudio.Shell.Interop;
+
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS.Waiting
+{
+    [Export(typeof(IImplicitlyActiveService))]
+    [Export(typeof(IOperationWaitIndicator))]
+    [AppliesTo(ProjectCapability.CSharpOrVisualBasic)]
+    internal partial class VisualStudioOperationWaitIndicator : AbstractMultiLifetimeComponent<VisualStudioOperationWaitIndicator.Instance>, IImplicitlyActiveService, IOperationWaitIndicator
```

Comment:
I'm not understanding why this implements AbstractMultiLifetimeComponent and why it needs to be loaded in and activated for every implicitly configuration.

From reading the code, this class basically does zero initialization except a GetService? Why can't that just be done when someone calls one of the methods that it implements? ie It doesn't seem like this needs any initialization at all.

Reply:
Yeah this should be a pure VS concept with no ties to the project system.  I'll fix this